src/tools/useLocalStorage.ts

```diff
@@ -17,20 +17,24 @@ export default function useLocalStorage(
   key: string,
   initialValue: string = ""
 ): UseLocalStorage {
+  const isWindowDefined = typeof window !== "undefined";
```

Comment:
Move this outside hook definition, I'd suggest somewhere like a `constants.ts` root file